### ISSUE TITLE
Cookie Banner handling Focus

### ISSUE THREAD

**Issue by iorgamgabriel** [mozilla-mobile/focus-android#7965]
@jeffreygee Thank you for the design !

**1.Settings part** 
This is the GeckoView interface for Cookie banner 
>
     * Represents settings options for cookie banner handling.
    enum class CookieBannerHandlingMode(val mode: Int) {
      
         * The feature is turned off and cookie banners are not handled
        DISABLED(0),

         * Reject cookies if possible
        REJECT_ALL(1),

         * Reject cookies if possible. If rejecting is not possible, accept cookies
        REJECT_OR_ACCEPT_ALL(2),
    }
<filepath>

Synthesising the Figma design we will have the following 3 scenarios:

1)If the 'Auto cookie banner consent' is not checked 'Further reduce cooke banners' item will be invisible and value for Cookie banner conform GeckoView interface will be   DISABLED(0).
2) If the 'Auto cookie banner consent' is checked 'Further reduce cooke banners' item will be visible and unchecked the value for Cookie banner conform GeckoView interface will be   REJECT_ALL(1).
3)If the 'Auto cookie banner consent' is checked 'Further reduce cooke banners' item will be visible and checked the value for Cookie banner conform GeckoView interface will be  REJECT_OR_ACCEPT_ALL(2).

PS : If the user unchecked 'Auto cookie banner consent'  'Further reduce cooke banners' will become unchecked .
When the user enters first in the App the value will be DISABLED(0) first case.

https://user-images.githubusercontent.com/8118371/200337796-3f80d7c0-cd1b-4f59-aa82-985670617c6f.mp4

<img width="699" alt="Screenshot 2022-11-07 at 16 50 23" src="https://user-images.githubusercontent.com/8118371/200339931-96cd56b8-c66a-4d7b-a38e-471ec6ce97d4.png">

When the cookie banner option is disabled, the summary text for preference will be Off, and when the cookie banner option is reject_or_accept_all or reject_all, it will be On.

<img src="https://user-images.githubusercontent.com/8118371/203551408-f6883302-de07-4f9b-b015-16d3154df91d.jpg" width="300" height="600" /> 

This is the initial ticket https://bugzilla.mozilla.org/show_bug.cgi?id=1797561

**2.Cookie banner panel part**
   1)If the CookieBannerHandleMode is reject_all or reject_or_accept_all, I show the toggle. If the site has no exception, the toggle will be true.

<img src="https://user-images.githubusercontent.com/8118371/202170298-fb073c37-f876-4a7a-91d5-ebb0cbce2158.png" width="300" height="600" /> 

 2) If the CookieBannerHandleMode is disabled, the toggle and all the cookie banner section will be hidden.

<img src="https://user-images.githubusercontent.com/8118371/202171088-d413a06f-ff5e-4b60-bf33-6c9fcfb2f548.jpg" width="300" height="600" /> 

3)If the user sets the toggle to false, I'm going to add the site to the exception list with the value MODE_DISABLED.

<img src="https://user-images.githubusercontent.com/8118371/202171869-87dec253-a4d3-441c-991e-014b7d434721.png" width="300" height="600" /> 

4)If the site doesn't have a cookie banner and CookieBannerHandleMode is reject_all, or accept_all, I will show the section but not the toggle.

<img src="https://user-images.githubusercontent.com/8118371/202172562-08319d20-9384-428c-be70-44b939fbfffb.png" width="300" height="600" />

**Comment by iorgamgabriel** [comment]
This is what I should implement in Focus ? An entry in Focus Settings under Block cookies that will redirect the user to this screen?
<img src="https://user-images.githubusercontent.com/8118371/199506805-dd4932d3-3438-4909-a923-f3a2ba2a541d.jpg" width="300" height="600" /> 
<img src="https://user-images.githubusercontent.com/8118371/199504825-242150ce-56df-47e9-b651-4f9a17a545fd.jpg" width="300" height="600" /> 
<img src="https://user-images.githubusercontent.com/8118371/199506809-330291a5-58ac-4461-99f8-61208d47d879.jp" width="300" height="600" />

**Comment by Amejia481** [comment]
👋🏽 the Fenix UI still hasn't finalized I'm not sure about Focus, you can find the mockups here https://www.figma.com/file/o3TMGgSnpGNvA9GXb7FUu6/Auto-Cookie-Banners-(Focus)?node-id=603%3A17020 @jeffreygee can help to clarify.

**Comment by iorgamgabriel** [comment]
ok Thank you @Amejia481

**Comment by iorgamgabriel** [comment]
https://www.figma.com/file/o3TMGgSnpGNvA9GXb7FUu6/Auto-Cookie-Banners-(Focus)?node-id=603%3A17020
On the second screen, the new cookie banner cfr will appear instead of the track protection cfr ?

<img src="https://user-images.githubusercontent.com/8118371/199535509-e1dab240-4d29-4682-8b42-eaac2c196f4a.jpeg" width="300" height="600" />

**Comment by jeffreygee** [comment]
Sorry, I didn't realize these were being built out already. But yes, it will follow this UI on this [Figma](https://www.figma.com/file/o3TMGgSnpGNvA9GXb7FUu6/Auto-Cookie-Banners-(Focus)?node-id=250%3A7861). I will update this tomorrow with the Light/Dark modes

We're currently doing testing on the content, but that will just be copy changes if anything.

@iorgamgabriel - to answer your question, discussed this with @rtestard and I believe the decision is that cookie banner CFR wouldn't be triggered until onboarding was complete. I'll let Romain confirm here.

**Comment by iorgamgabriel** [comment]
<img src="https://user-images.githubusercontent.com/8118371/199919311-f6203f34-5a18-4d5a-ae6b-9e72ef7adc60.jpg" width="300" height="600" /> 
I think we shouldn't put an arrow next to the "Auto cookie banner consent" entry, because the arrow is used only for "Block cookies," which will trigger a pop-up. 
In your [Figma](https://www.figma.com/file/o3TMGgSnpGNvA9GXb7FUu6/Auto-Cookie-Banners-(Focus)?node-id=250%3A7861) "Site Permissions" entry has an arrow, but in the app it doesn't. @rtestard @jeffreygee

**Comment by rtestard** [comment]
> 

Yes, the trigger should be per spec:
After completion of new users onboarding, on the first instance of a cookie banner that gets cleared, a panel opens off of the shield button...

**Comment by iorgamgabriel** [comment]
@rtestard But if the site has trackers the cfr for tracking protection will appear in the same place like this one .

**Comment by rtestard** [comment]
@iorgamgabriel Agreed that "Site permissions" should the same pattern as "Auto cookie banner consent"
"Studies" has the same pattern and also no arrow.
@jeffreygee should they all have an arrow or none?

**Comment by iorgamgabriel** [comment]
The GUI is wrong for settings last screen  [Figma](https://www.figma.com/file/o3TMGgSnpGNvA9GXb7FUu6/Auto-Cookie-Banners-(Focus)?node-id=250%3A7861) . This is the GeckoView interface : 

/**
 * Represents settings options for cookie banner handling.
 */
enum class CookieBannerHandlingMode(val mode: Int) {
    /**
     * The feature is turned off and cookie banners are not handled
     */
    DISABLED(0),

    /**
     * Reject cookies if possible
     */
    REJECT_ALL(1),

    /**
     * Reject cookies if possible. If rejecting is not possible, accept cookies
     */
    REJECT_OR_ACCEPT_ALL(2),
}
You can have only one of these options , with this GUI you can have   REJECT_ALL(1) and REJECT_OR_ACCEPT_ALL(2)
the Gecko Engine interface doesn't allow this. I think we can use something like in Fenix.
@jeffreygee @amedyne @rtestard

**Comment by iorgamgabriel** [comment]
@jeffreygee Thank you for the design !
This is the GeckoView interface for Cookie banner 
>
     * Represents settings options for cookie banner handling.
    enum class CookieBannerHandlingMode(val mode: Int) {
      
         * The feature is turned off and cookie banners are not handled
        DISABLED(0),

         * Reject cookies if possible
        REJECT_ALL(1),

         * Reject cookies if possible. If rejecting is not possible, accept cookies
        REJECT_OR_ACCEPT_ALL(2),
    }
<filepath>

1)If the 'Auto cookie banner consent' is not checked 'Further reduce cooke banners' item will be invisible and value for Cookie banner conform GeckoView interface will be   DISABLED(0).
2) If the 'Auto cookie banner consent' is checked 'Further reduce cooke banners' item will be visible and unchecked the value for Cookie banner conform GeckoView interface will be   REJECT_ALL(1).
3)If the 'Auto cookie banner consent' is checked 'Further reduce cooke banners' item will be visible and checked the value for Cookie banner conform GeckoView interface will be  REJECT_OR_ACCEPT_ALL(2).
PS : If the user unchecked 'Auto cookie banner consent'  'Further reduce cooke banners' will become unchecked .

https://user-images.githubusercontent.com/8118371/200337796-3f80d7c0-cd1b-4f59-aa82-985670617c6f.mp4

@rtestard @jeffreygee @amedyne Is it correct?

**Comment by iorgamgabriel** [comment]
<img width="429" alt="Screenshot 2022-11-08 at 14 33 16" src="https://user-images.githubusercontent.com/8118371/200564874-1857286d-1f6c-413f-b32e-3821e076f1ff.png">
What state should the cookie banner be in when the user has enabled it? REJECT_ALL(1) or REJECT_OR_ACCEPT_ALL(2)

The icon on the panel should be different if the cookie banner is enabled or disabled?
@rtestard @jeffreygee @amedyne

**Comment by jeffreygee** [comment]
@rtestard - agreed. I've just put in a request for this with the design systems team

**Comment by jeffreygee** [comment]
![CookieOff](https://user-images.githubusercontent.com/93741573/203581417-453de57b-8ee2-4146-9d56-f6984194a130.png)

Making a note here that we've received the new cookie icon for disabled and no detected. Updated in the [figma file.](https://www.figma.com/file/o3TMGgSnpGNvA9GXb7FUu6/Auto-Cookie-Banners-(Focus)?node-id=603%3A17020)

**Comment by lobontiumira** [comment]
Verified on a local debug build that:
* the "Cookie Banner Reduction" option is available in Settings - Privacy & Security - Cookies and Site Data,
* the "Further reduce cookie banners" option is displayed only when the "Cookie Banner Reduction" toggle is enabled,
* the "Further reduce cookie banners" option can be checked/unchecked, and it is working correctly.

![cookie banner](https://user-images.githubusercontent.com/40694158/204010644-2a44be0b-498a-4f53-a987-559af19b3682.png)

**Note**:
* enabling the "Cookie Banner Reduction" toggle, and the "Further reduce cookie banners" option, does not dismiss the cookie banners of ALL the websites from the Cookie Banner list - I'll file new issues on Bugzilla, as soon as I identify all the problematic websites.
I'll remove the QAReady label for now, but leave this ticket opened, until I've verified the entire Cookie Banner list.
